azurelinuxagent/ga/monitor.py

```diff
@@ -177,11 +180,35 @@ def collect_and_send_events(self):
         except ProtocolError as e:
             logger.error("{0}", e)
 
+    def send_daily_heartbeat(self):
+        """
+        Send the daily heartbeat containing the installed and current agent versions.
+        Send once during the first 30 minutes of the day.
+        """
+        if datetime.datetime.time() <= datetime.time(0, minute=30):
```

Comment:
I thought about that. In fact, I had originally written a timestamp to disk so that it was emitted only once every 24 hours. But the small error we have with the simpler code did not, in my mind, justify the increased complexity of maintaining the timestamp.